File: poet-spring-boot-starter/src/main/java/com/skypyb/poet/spring/boot/core/model/DefaultPoetAnnex.java
package com.skypyb.poet.spring.boot.core.model;

public class DefaultPoetAnnex implements PoetAnnex {

    private String name;

    private String realName;

    private String suffix;

    private String key;

    private Long length;

    /**
     * 没有 realName&length
     */
    public static DefaultPoetAnnex of(Navigation n) {

        String name = n.getName();

        DefaultPoetAnnex defaultPoetAnnex = new DefaultPoetAnnex();
        defaultPoetAnnex.setName(name);
        defaultPoetAnnex.setKey(n.getPath());

        int suffixPoint = name.lastIndexOf(".");
        if (suffixPoint > 0) {
            //后缀名保存时不保留 "."
            defaultPoetAnnex.setSuffix(name.substring(suffixPoint + 1));
        }

        return defaultPoetAnnex;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public void setRealName(String realName) {
        this.realName = realName;
    }


    public void setSuffix(String suffix) {
        this.suffix = suffix;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public void setLength(Long length) {
        this.length = length;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public String getRealName() {
        return realName;
    }


    @Override
    public String getSuffix() {
        return suffix;
    }

    @Override
    public String getKey() {
        return key;
    }


    @Override
    public Long getLength() {
        return length;
    }

}

File: poet-spring-boot-starter/src/main/java/com/skypyb/poet/spring/boot/core/model/DefaultPoetAnnex.kt
package com.skypyb.poet.spring.boot.core.model

class DefaultPoetAnnex : PoetAnnex {

    private var name: String? = null
    private var realName: String? = null
    private var suffix: String? = null
    private var key: String? = null
    private var length: Long? = null


    override fun setName(name: String) {
        this.name = name
    }

    override fun setRealName(realName: String) {
        this.realName = realName
    }

    override fun setSuffix(suffix: String) {
        this.suffix = suffix
    }

    override fun setKey(key: String) {
        this.key = key
    }

    override fun setLength(length: Long) {
        this.length = length
    }

    override fun getName(): String {
        return name!!
    }

    override fun getRealName(): String {
        return realName!!
    }

    override fun getSuffix(): String {
        return suffix!!
    }

    override fun getKey(): String {
        return key!!
    }

    override fun getLength(): Long {
        return length!!
    }

    companion object {
        /**
         * 没有 realName&length
         */
        @JvmStatic
        fun of(n: Navigation): DefaultPoetAnnex {
            val name = n.name
            val defaultPoetAnnex = DefaultPoetAnnex()
            defaultPoetAnnex.setName(name!!)
            defaultPoetAnnex.setKey(n.path!!)
            val suffixPoint = name.lastIndexOf(".")
            if (suffixPoint > 0) { //后缀名保存时不保留 "."
                defaultPoetAnnex.setSuffix(name.substring(suffixPoint + 1))
            }
            return defaultPoetAnnex
        }
    }
}